File: 0645-set-mismatch/0645-set-mismatch.cpp
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        int n = nums.size();
        vector<int> occ (n, 0);
        for (int i = 0; i < n; i++) {
            occ[nums[i] - 1]++;
        }
        int missingNum, occTwice;
        for (int i = 0; i < n; i++) {
            if (occ[i] == 0) {
                missingNum = i + 1;
            } else if (occ[i] == 2) {
                occTwice = i + 1;
            }
        }
        return {occTwice, missingNum};
    }
};

File: 0645-set-mismatch/0645-set-mismatch.java
class Solution {
    public int[] findErrorNums(int[] nums) {
        int i = 0;
        while (i < nums.length) {
            int correctPos = nums[i] - 1;
            if (correctPos != i && nums[correctPos] != correctPos + 1) {
                int temp = nums[correctPos];
                nums[correctPos] = nums[i];
                nums[i] = temp;
            } else {
                i++;
            }
        }
        
        int[] ans = new int[2];
        for (int j = 0; j < nums.length; j++) {
            if (j + 1 != nums[j]) {
                ans[0] = nums[j];
                ans[1] = j + 1;
            }
        }
        
        return ans;
    }
}